incubator/buzzfeed-sso/README.md

```diff
@@ -0,0 +1,183 @@
+
+```bash
+$ helm install --name my-release stable/buzzfeed-sso
+```
+
+The command deploys SSO on the Kubernetes cluster using the default configuration. The [configuration](#configuration) section lists the parameters that can be configured during installation.
+
+This chart has required variables, see [Configuration](#configuration).
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` deployment:
+
+```bash
+$ helm delete --purge my-release
+```
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following table lists the configurable parameters of the SSO chart and their default/required values.
+
+Parameter | Description | Default
+--- | --- | ---
+`namespace` | namespace to use | `default`
+`emailDomain` | the sso email domain for authentication | REQUIRED
+`rootDomain` | the parent domain used for protecting your backends | REQUIRED
+`auth.domain` | the auth domain used for OAuth callbacks | REQUIRED
+`auth.replicaCount` | desired number of auth pods | `1`
+`auth.resources` | resource limits and requests for auth pods | `{ limits: { memory: "256Mi", cpu: "200m" }}`
+`auth.nodeSelector` | node selector logic for auth pods | `{}`
+`auth.tolerations` | resource tolerations for auth pods | `{}`
+`auth.affinity` | node affinity for auth pods | `{}`
+`auth.service.type` | type of auth service to create | `ClusterIP`
+`auth.service.port` | port for the http auth service | `80`
+`auth.secret` | secrets to be generated randomly with `openssl rand -base64 32 | head -c 32`. | REQUIRED if `auth.customSecret` is not set
+`auth.tls` | tls configuration for central sso auth ingress. | `{ secretName: "sso-auth-tls-secret" }`
+`auth.customSecret` | the secret key to reuse (avoids secret creation via helm) | REQUIRED if `auth.secret` is not set
+`proxy.virtualHost` | wildcard domain for redirecting SSO to the backends | REQUIRED
+`proxy.providerUrlInternal` | url for split dns deployments |
+`proxy.cluster` | the cluster name for SSO | `dev`
+`proxy.replicaCount` | desired number of proxy pods | `1`
+`proxy.resources` | resource limits and requests for proxy pods | `{ limits: { memory: "256Mi", cpu: "200m" }}`
+`proxy.nodeSelector` | node selector logic for proxy pods | `{}`
+`proxy.tolerations` | resource tolerations for proxy pods | `{}`
+`proxy.affinity` | node affinity for proxy pods | `{}`
+`proxy.service.type` | type of proxy service to create | `ClusterIP`
+`proxy.service.port` | port for the http proxy service | `80`
+`proxy.secret` | secrets to be generated randomly with `openssl rand -base64 32 | head -c 32 | base64`. | REQUIRED if `proxy.customSecret` is not set
+`proxy.customSecret` | the secret key to reuse (avoids secret creation via helm) | REQUIRED if `proxy.secret` is not set
+`provider.google` | the Oauth provider to use (only Google support for now) | REQUIRED
+`provider.google.adminEmail` | the Google admin email | REQUIRED
```

Comment:
@darioblanco it can be useful to have this value being optional. If not specified, we would skip everything related to the service account in the auth deployment for people who don't want to setup group access control

Reply:
Now the environment variable for the google admin email will be skipped if not specified. I did not know that it wasn't required, thanks for the feedback!